pkg/image/importer/client.go

```diff
@@ -109,17 +110,33 @@ type repositoryRetriever struct {
 	redirect map[url.URL]*url.URL
 }
 
-func (r *repositoryRetriever) Repository(ctx gocontext.Context, registry *url.URL, repoName string, insecure bool) (distribution.Repository, error) {
+var _ RepositoryRetriever = &repositoryRetriever{}
+
+func (r *repositoryRetriever) Repository(ctx gocontext.Context, registryURL *url.URL, repoName string, insecure bool) (distribution.Repository, error) {
 	named, err := reference.ParseNamed(repoName)
 	if err != nil {
 		return nil, err
 	}
 
-	t := r.context.Transport
-	if insecure && r.context.InsecureTransport != nil {
-		t = r.context.InsecureTransport
+	tlsConfig := &tls.Config{
+		MinVersion:               tls.VersionTLS10,
+		PreferServerCipherSuites: true,
+		InsecureSkipVerify:       insecure,
+	}
+	if err := registry.ReadCertsDirectory(tlsConfig, filepath.Join(registry.CertsDir, registryURL.Host)); err != nil {
```

Comment:
It would be better if this is passed in as an opaque interface to the context.  Call it `CertRetriever` or something, and have it have a method like `TransportFor(host) (http.RoundTripper, error)`.  That would make this better for testing.

In the implementation, you'll want to fallback to the defaults defined above.

Reply:
Or TransportRetriever